Add linting of web features file

jcscottiii opened the pull request:
Checks:
- If the file meets the correct format
- If the files referenced in the file are present.

This is part of a multi PR plan
- PR 1 - Parse META.yml files - #43394
- PR 2 - Add lint of META.yml files - #43395 
- PR 3 - Parse WEB_FEATURES.yml files - #43403
- PR 4 - Add lint of WEB_FEATURES.yml files (This PR) - #43401

jcscottiii committed:
Add linting of web features file

foolip commented on tools/lint/lint.py:
```diff
@@ -674,6 +675,27 @@ def check_meta_file(repo_root: Text, path: Text, f: IO[bytes]) -> List[rules.Err
+    for feature in web_features_file.features:
+        if isinstance(feature.files, list):
+            for file in feature.files:
+                if not file.match_files(files_in_directory):
```
If I'm reading everything correctly, the lint will fail also in the case of a pattern that doesn't match anything. I think that's important because it'll be quite easy to write a pattern that ends up matching nothing after some renames, and the lint should then fail. 👍

foolip commented on tools/lint/tests/test_file_lints.py:
```diff
@@ -809,6 +809,108 @@ def test_invalid_meta_file():
+features:
+- name: feature1
+  files:
+  - file*.txt
```
Can you also add a test with a pattern that doesn't match? Ideally one pattern that matches something and another that matches something in the same test, so that we know that the check is at the level of the pattern, not that all files/patterns together matches something.

jcscottiii replied:
Added! Will merge after clean CI run.

jcscottiii committed:
Add test to assert failure to match a pattern

comment: https://github.com/web-platform-tests/wpt/pull/43401#discussion_r1430364048
